WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift

```diff
@@ -29,25 +34,74 @@ import Foundation
     }
 
+    public override func sizeThatFits(size: CGSize) -> CGSize {
+        // Vertical and horizontal margins
+        let hMargin = descriptionLabel.frame.minX
+        let vMargin = descriptionLabel.frame.minY
+
+        let width = UIDevice.isPad() ? contentIPadWidthConstraint!.constant : size.width
+        let innerWidth = width - (hMargin * 2)
+        let adjustedSize = CGSize(width:innerWidth, height:CGFloat.max)
+
+        var height = UIDevice.isPad() ? contentIPadTopConstraint!.constant : 0
+        height += descriptionView.frame.minY
+        height += vMargin
+        height += descriptionLabel.sizeThatFits(adjustedSize).height
+        height += descriptionBottomConstraint.constant
+        height += followCountLabel.sizeThatFits(adjustedSize).height
+        height += followCountBottomConstraint.constant;
+
+        return CGSize(width: size.width, height: height)
+    }
+
 
    // MARK: - Configuration
 
     public func configureHeader(topic: ReaderAbstractTopic) {
-        // TODO: Wire up actual display when supported in core data
-        avatarImageView.setImageWithURL(NSURL(), placeholderImage: UIImage(named: defaultBlavatar))
-        titleLabel.text = topic.title
-        detailLabel.text = "site.com"
-        if topic.following {
+        assert(topic.isKindOfClass(ReaderSiteTopic), "Topic must be a site topic")
+
+        let siteTopic = topic as! ReaderSiteTopic
+
+        let url = NSURL(string: siteTopic.siteBlavatar)!
+        avatarImageView.setImageWithURL(url, placeholderImage: UIImage(named: defaultBlavatar))
+        titleLabel.text = siteTopic.title
+        detailLabel.text = NSURL(string: siteTopic.siteURL)?.host
+        if siteTopic.following {
             WPStyleGuide.applyReaderStreamHeaderFollowingStyle(followButton)
         } else {
             WPStyleGuide.applyReaderStreamHeaderNotFollowingStyle(followButton)
         }
 
-        followCountLabel.text = "100 followers"
-        descriptionLabel.text = "Just another WordPress site"
-//        var attributes = WPStyleGuide.readerStreamHeaderDescriptionAttributes()
-//        var attributedText = NSAttributedString(string: topic.description, attributes: attributes)
-//        descriptionLabel.attributedText = attributedText
+        descriptionLabel.attributedText = attributedSiteDescriptionForTopic(siteTopic)
+
+        if descriptionLabel.attributedText?.length > 0 {
+            // Bottom and top margins should match.
+            descriptionBottomConstraint.constant = descriptionLabel.frame.minY
+        } else {
+            descriptionBottomConstraint.constant = 0
+        }
+
+
+        followCountLabel.text = formattedFollowerCountForTopic(siteTopic)
+    }
+
+    func formattedFollowerCountForTopic(topic:ReaderSiteTopic) -> String {
+        let numberFormatter = NSNumberFormatter()
+        numberFormatter.groupingSeparator = NSLocale.currentLocale().objectForKey(NSLocaleGroupingSeparator) as! String
```

Comment:
I think you can skip this line as it is the default behavior

<img width="737" alt="screen shot 2015-09-23 at 11 01 21" src="https://cloud.githubusercontent.com/assets/8739/10041169/80f6df3c-61e2-11e5-8f18-1f9fe8c9c000.png">